src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpSyntaxFormattingOptions.cs

```diff
@@ -0,0 +1,222 @@
+
+using System;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+
+namespace Microsoft.CodeAnalysis.CSharp.Formatting
+{
+    [Flags]
+    internal enum SpacePlacement
+    {
+        IgnoreAroundVariableDeclaration = 1,
+        AfterMethodDeclarationName = 1 << 1,
+        BetweenEmptyMethodDeclarationParentheses = 1 << 2,
+        WithinMethodDeclarationParenthesis = 1 << 3,
+        AfterMethodCallName = 1 << 4,
+        BetweenEmptyMethodCallParentheses = 1 << 5,
+        WithinMethodCallParentheses = 1 << 6,
+        AfterControlFlowStatementKeyword = 1 << 7,
+        WithinExpressionParentheses = 1 << 8,
+        WithinCastParentheses = 1 << 9,
+        BeforeSemicolonsInForStatement = 1 << 10,
+        AfterSemicolonsInForStatement = 1 << 11,
+        WithinOtherParentheses = 1 << 12,
+        AfterCast = 1 << 13,
+        BeforeOpenSquareBracket = 1 << 14,
+        BetweenEmptySquareBrackets = 1 << 15,
+        WithinSquareBrackets = 1 << 16,
+        AfterColonInBaseTypeDeclaration = 1 << 17,
+        BeforeColonInBaseTypeDeclaration = 1 << 18,
+        AfterComma = 1 << 19,
+        BeforeComma = 1 << 20,
+        AfterDot = 1 << 21,
+        BeforeDot = 1 << 22,
+    }
+
+    [Flags]
+    internal enum NewLinePlacement
+    {
+        BeforeMembersInObjectInitializers = 1,
+        BeforeMembersInAnonymousTypes = 1 << 1,
+        BeforeElse = 1 << 2,
+        BeforeCatch = 1 << 3,
+        BeforeFinally = 1 << 4,
+        BeforeOpenBraceInTypes = 1 << 5,
+        BeforeOpenBraceInAnonymousTypes = 1 << 6,
+        BeforeOpenBraceInObjectCollectionArrayInitializers = 1 << 7,
+        BeforeOpenBraceInProperties = 1 << 8,
+        BeforeOpenBraceInMethods = 1 << 9,
+        BeforeOpenBraceInAccessors = 1 << 10,
+        BeforeOpenBraceInAnonymousMethods = 1 << 11,
+        BeforeOpenBraceInLambdaExpressionBody = 1 << 12,
+        BeforeOpenBraceInControlBlocks = 1 << 13,
+        BetweenQueryExpressionClauses = 1 << 14
+    }
+
+    [Flags]
+    internal enum IndentationPlacement
+    {
+        Braces = 1,
+        BlockContents = 1 << 1,
+        SwitchCaseContents = 1 << 2,
+        SwitchCaseContentsWhenBlock = 1 << 3,
+        SwitchSection = 1 << 4
+    }
+
+    internal sealed class CSharpSyntaxFormattingOptions : SyntaxFormattingOptions
+    {
+        public readonly SpacePlacement Spacing;
+        public readonly BinaryOperatorSpacingOptions SpacingAroundBinaryOperator;
+        public readonly NewLinePlacement NewLines;
+        public readonly LabelPositionOptions LabelPositioning;
+        public readonly IndentationPlacement Indentation;
+        public readonly bool WrappingKeepStatementsOnSingleLine;
+        public readonly bool WrappingPreserveSingleLine;
+
+        public CSharpSyntaxFormattingOptions(
+            bool useTabs,
+            int tabSize,
+            int indentationSize,
+            string newLine,
```

Comment:
i feel like teh above 4 often go hand in hand.  any thoughts on combining to a basic formatting options struct?